### PR TITLE
Fix npm publish configuration

### PR DESCRIPTION
**Pull request by AlexKolpa**
Instead of just publishing and using the source as our entry point, it now checks with eslint first, then transpiles it and publishes the transpiled code instead. Additionally fixed the npm ignores and git ignores.

@Magnetme/developers ready for review

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,13 @@
 {
   "name": "@magnet.me/consultant",
-  "version": "1.0.0",
+  "version": "1.0.2",
   "description": "A library to register services to Consul and listen for property changes.",
-  "main": "src/index.js",
+  "main": "out/index.js",
```
Comment (rogierslag) [inline package.json:5]: traditionally its `lib`

**Review by rogierslag — APPROVED** [review]

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.idea/
+.git/
```
Comment (rogierslag) [inline .npmignore:2]: If you want to be really nice to the users of your lib, exclude everything but the final `index.js`. There is no need they should bundle your readme in their distribution

Comment (AlexKolpa) [reply .npmignore:2]: >There is no need they should bundle your readme in their distribution

>The following paths and files are never ignored, so adding them to .npmignore is pointless: README (and its variants)

I'll have a look at what I can clean up though!

Comment (rogierslag) [reply .npmignore:2]: Example is the `src` folder then

File: package.json (hunk 1 above)
Comment (AlexKolpa) [reply package.json:5]: Interesting. So far I've only seen `out` used here at magnet.me, so I'd rather keep it consistent with our own stack.

**Pull request merged**